src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyoutViewModel.cs

```diff
@@ -55,7 +55,7 @@ public string IdentifierText
             {
                 if (value != _session.ReplacementText)
                 {
-                    _session.ApplyReplacementText(value, propagateEditImmediately: true);
```

Comment:
The second place causes ArugmentOutofRangeException.
When the identifier in the text box gets changed, it would [update](https://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/InlineRename/InlineRenameSession.OpenTextBufferManager.cs,287) the text selection in the editor.

This might cause an Exception because text selection might also get changed when the replacement text changes. 
(Like, if you type `F` when rename `Field123`, the replacement text is `F`, and text selection becomes zero length)
We are still using a out-off date state https://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/InlineRename/AbstractInlineRenameUndoManager.cs,143 to update the selection in the editor.

Reply:
This is triggered by the binding back to the UI. And we only know replacement text gets changed.
IMO we should not try to update selection at this point. Later it would be updated by another event handler of the control https://github.com/dotnet/roslyn/blob/d7e010bbe5b1d37837417fc5e79ecb2fd9b7b487/src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyout.xaml.cs#L210